### PR TITLE
Prepare packages for publishing

### LINKED ISSUE #533: Distribute contract artifacts
We want to make it simple for developers to third party developers to build on top of the Vault and known Pools.

This is a tentative list of basic use cases we want to support:
 - write smart contracts that interact with the Vault (via `IVault`) or the different Pools
 - setup a local environment, complete with an Authorizer and Pool factories
 - reuse library code (error codes, base pools, etc.)

#467 is a requirement for this, as a single package would not serve all of these needs (and others we may have).

### PR DESCRIPTION
**Pull request by nventuro**
This prepares most Solidity packages for them to be released. 

It adds the `files` field to the different `package.json` files, as well as minimal readmes for each of them so there is _something_ to show on npm. I've updated the repository links to `balancer-v2-monorepo`, in preparation for the renaming of this repository.

It also sets up dependencies between packages instead of dev-dependencies, meaning a user that e.g. installs `pool-utils` will also get `vault` and `solidity-utils` automatically. You will notice that the dependencies don't list semver ranges but rather use the `workspace:*` syntax: when running `yarn pack` or `yarn npm publish` these are automatically updated to the current version.

I've bumped all packages we intend to publish immediately (deployments, vault, pool-weighted, pool-utils, solidity-utils) to 1.0.0: pool-stable and asset-manager-utils remain in 0.1.0 meaning they are not quite ready to have an initial version out.

I've additionally installed the `yarn version` plugin, which we can use to track which packages are pending a version bump as we make commits to the repository. You can read more about this plugin [here](https://yarnpkg.com/features/release-workflow).

Closes #533

### CODE REVIEW

File: pkg/pool-weighted/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# <img src="../../logo.svg" alt="Balancer" height="128px">
+
+# Balancer V2 Weighted Pools
+
+[![NPM Package](https://img.shields.io/npm/v/@balancer-labs/v2-pool-weighted.svg)](https://www.npmjs.org/package/@balancer-labs/v2-pool-weighted)
+
+This package contains the source code of Balancer V2 Weighted Pools, that is, Pools that swap tokens by enforcing a Constant Weighted Product invariant.
```
Comment (EndymionJkb) [inline pkg/pool-weighted/README.md:7]:
```suggestion
This package contains the source code for Balancer V2 Weighted Pools, that is, Pools that swap tokens by enforcing a Constant Weighted Product invariant.
```

File: pkg/solidity-utils/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# <img src="../../logo.svg" alt="Balancer" height="128px">
+
+# Balancer V2 Solidity Utilities
+
+[![NPM Package](https://img.shields.io/npm/v/@balancer-labs/v2-solidity-utils.svg)](https://www.npmjs.org/package/@balancer-labs/v2-solidity-utils)
+
+This package contains Solidity utilities and libraries used when developing Balancer V2 contracts. Many design decisions and trade-offs have been made in the context of Balancer V2's requirements and constraints (such as reduced bytecode size), which may make these libraries not suitable for other projects.
```
Comment (EndymionJkb) [inline pkg/solidity-utils/README.md:7]:
```suggestion
This package contains Solidity utilities and libraries used when developing Balancer V2 contracts. Many design decisions and trade-offs have been made in the context of Balancer V2's requirements and constraints (such as reduced bytecode size), which may make these libraries unsuitable for other projects.
```

File: pkg/vault/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+# <img src="../../logo.svg" alt="Balancer" height="128px">
+
+# Balancer V2 Vault
+
+[![NPM Package](https://img.shields.io/npm/v/@balancer-labs/v2-vault.svg)](https://www.npmjs.org/package/@balancer-labs/v2-vault)
+[![Docs](https://img.shields.io/badge/docs-%F0%9F%93%84-blue)](https://docs.balancer.fi/developers/smart-contracts/apis/vault)
+
+This package contains the source code of Balancer V2's main contract, the [`Vault`](./contracts/Vault.sol), as well as all [core interfaces](./contracts/interfaces).
+
+The [Vault interface](./contracts/interfaces/IVault.sol) itself is required for multiple tasks, including swaps, joins and exits, but some scenario require using a few additional ones. Particularly useful interfaces include:
```
Comment (EndymionJkb) [inline pkg/vault/README.md:10]:
```suggestion
The [Vault interface](./contracts/interfaces/IVault.sol) itself is required for multiple tasks, including swaps, joins and exits, but some scenarios require functionality provided through other contracts. Particularly useful interfaces include:
```

File: pkg/vault/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+### Installation
+
+```console
+$ npm install @balancer-labs/v2-vault
+```
+
+### Usage
+
+The main purpose if this package is to provide the Solidity interfaces of core components, which are found in the [`/contracts/interfaces`](./contracts/interfaces] directory. The Vault contract itself however is still useful for testing.
```
Comment (EndymionJkb) [inline pkg/vault/README.md:26]:
```suggestion
The main purpose of this package is to provide the Solidity interfaces of core components, which are found in the [`/contracts/interfaces`](./contracts/interfaces] directory. The Vault contract itself is also useful for testing.
```

**Review by EndymionJkb — CHANGES_REQUESTED** [review]

**Comment by nventuro** [comment]
This should be now ready to publish.

File: pkg/pool-weighted/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      IPriceOracle.OracleAverageQuery[] memory queries = new IPriceOracle.OracleAverageQuery[](1);
+
+      // Average price over the last hour
+      queries[0] = IPriceOracle.OracleAverageQuery({
+        variable: IPriceOracle.Variable.PAIR_PRICE,
+        secs: 3600,
+        ago: 0
+      });
+
+      uint256[] memory results = oracle.getTimeWeightedAverage(queries);
```
Comment (EndymionJkb) [inline pkg/pool-weighted/README.md:47]: Should this check that the oracle has been fully initialized? (Or at least have a comment that it's only guaranteed to work then?)

File: pkg/pool-weighted/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+### Installation
+
+```console
+$ npm install @balancer-labs/v2-pool-weighted
+```
+
+### Usage
+
+This package can be used in multiple ways, including interacting with already deployed Pools, performing local testing, or even create new Pool types that also use the Constant Weighted Product invariant.
```
Comment (EndymionJkb) [inline pkg/pool-weighted/README.md:23]:
```suggestion
This package can be used in multiple ways, including interacting with already deployed Pools, performing local testing, or even creating new Pool types that also use the Constant Weighted Product invariant.
```

**Review by EndymionJkb — CHANGES_REQUESTED** [review]

**Pull request merged**